docs/hugo/content/design/ADR-2023-02-Helm-Chart-Size-Limitations.md

```diff
@@ -0,0 +1,203 @@
+3. We embed the CRDs into the operator pod and have the operator pod manage them on startup.
+
+For all of the above options, we would still maintain a Helm chart for the operator itself (the deployment, service account, etc).
+
+Of the above options, I believe embedding the CRDs into the operator pod is the best option. Here are some pros/cons for it.
+
+**Pros**:
+- Helm wasn't really getting us a lot for managing CRDs anyway, and embedding the CRDs into the operator itself
+  means that we can guarantee the right versions are installed for the corresponding operator version (unlike all
+  the chart-splitting options above)
+- We're able to continue to offer an easy-to-use Helm chart that provides the entire ASO feature set.
+- Because it's integrated into the operator we could add safeguards and capabilities to this "CRD reconciliation" 
+  for both deprecation or uninstall cases over and above what we could easily do in Helm.
+- More easily enables [#1433](https://github.com/Azure/azure-service-operator/issues/1433) as compared to a single 
+  monolithic Helm chart. We could offer configuration options either via a ConfigMap or a CRD that _is_ included in the
+  Helm chart. That single included CRD would be something like `InstalledResourceDefinitions` and support specifying what groups
+  or individual resources would be installed.
+
+**Cons**:
+- CRD installation would _probably_ be a required pre-step and if it failed the operator would stop running, which means
+  existing resources wouldn't be reconciled until the deployment was rolled back. We could possibly mitigate this by making
+  CRD installation failure nonfatal, but that has other complications.
+- Operator needs permission to CRUD CRDs, which it previously didn't have.
+- Uninstalling the Helm chart will uninstall the operator pod but will not uninstall the CRDs. This is somewhat mitigated
+  by the fact that uninstalling CRDs is very dangerous and/or impossible if there are existing resources of that CRD type
+  even in the current ASO chart.
+
+Another possible option is using `asoctl`. Since that involves writing pretty much the same code as would be required to 
+manage the CRDs from the operator, and would preclude installing ASO as an addon in AKS easily (since they want everything
+Helm deployable), it seems to make sense to manage the CRDs from the operator rather than from `asoctl`.
+
+### Digging in
+We will introduce a new CRD `InstalledResourceDefinitions.serviceoperator.azure.com`, with the following structure:
+```go
+// +kubebuilder:object:root=true
+type InstalledResourceDefinitionsList struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ListMeta `json:"metadata,omitempty"`
+    Items           []InstalledResourceDefinitions `json:"items"`
+}
+
+type InstalledResourceDefinitionsSpec struct {
+    // +kubebuilder:validation:Required
+    // Pattern defines the set of CRDs which should be installed. Currently only "*" (all resources) is supported.
```

Comment:
Suggest indicating where this might go in future.

```suggestion
    // Pattern defines the set of CRDs which should be installed. 
    // Currently only "*" (all resources) is supported.
    // Future expansion will allow "_group_.*" to install all resources from a group, 
    // as well as "_group_._kind_" to install a specific one
```

Reply:
Added